src/components/listitems/ListItemHeader.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { ComponentProps, useCallback, useMemo } from "react";
+import { View } from "react-native";
+import {
+  IOListItemStyles,
+  IOListItemVisualParams,
+  IOStyles,
+  useIOTheme
+} from "../../core";
+import { WithTestID } from "../../utils/types";
+import { IOIcons, Icon } from "../icons";
+import { H6 } from "../typography";
+import { ButtonLink, IconButton } from "../buttons";
+import { Badge } from "../badge";
+
+type ButtonLinkActionProps = {
+  type: "buttonLink";
+  componentProps: ComponentProps<typeof ButtonLink>;
+};
+
+type IconButtonActionProps = {
+  type: "iconButton";
+  componentProps: ComponentProps<typeof IconButton>;
+};
+
+type BadgeProps = {
+  type: "badge";
+  componentProps: ComponentProps<typeof Badge>;
+};
+
+type EndElementProps =
+  | ButtonLinkActionProps
+  | IconButtonActionProps
+  | BadgeProps;
+
+export type ListItemHeader = WithTestID<{
+  label: string;
+  numberOfLines?: number;
+  icon?: IOIcons;
```

Comment:
`ListItemHeader` will be used in several screens. So we should be more open about custom visual attributes. In most cases, the `icon` prop with a default color will probably be enough. But in some specific cases, an `iconProps` object (like `componentProps`) might fit better. What do you think?

Reply:
if we keep the color fixed to `theme["icon-decorative"]` the only props accepted would be icon name and size... so you think it would be useful?

Reply:
My proposal is not to keep the color fixed (and not configurable), but to let the developer choose the icon color **if needed** (let's think about the specific pagoPA icon color we've already seen). I wouldn't make the size configurable. Of course `theme["icon-decorative"]` would be the correct fallback color if not specified.